gradoop-flink/src/main/java/org/gradoop/flink/io/impl/rdbms/functions/CreateEdges.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradoop.flink.io.impl.rdbms.functions;
+
+import java.util.List;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.types.Row;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.EdgeFactory;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.flink.io.impl.rdbms.connection.FlinkDatabaseInputHelper;
+import org.gradoop.flink.io.impl.rdbms.connection.RdbmsConfig;
+import org.gradoop.flink.io.impl.rdbms.metadata.MetaDataParser;
+import org.gradoop.flink.io.impl.rdbms.metadata.TableToEdge;
+import org.gradoop.flink.io.impl.rdbms.tuples.Fk1Fk2Props;
+import org.gradoop.flink.io.impl.rdbms.tuples.IdKeyTuple;
+import org.gradoop.flink.io.impl.rdbms.tuples.LabelIdKeyTuple;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+/**
+ * Creates Epgm edges from foreign key respectively n:m relations
+ *
+ */
+public class CreateEdges {
+
+  /**
+   * Creates EPGM edges from foreign key respectively n:m relations
+   *
+   * @param flinkConfig
+   *          Gradoop Flink configuration
+   * @param rdbmsConfig
+   *          Relational database configuration
+   * @param tablesToEdges
+   *          List of relations going to convert to edges
+   * @param vertices
+   *          Dataset of already created vertices
+   * @return Directed and undirected EPGM edges
+   * @throws Exception
+   */
+  public static DataSet<Edge> create(GradoopFlinkConfig flinkConfig, RdbmsConfig rdbmsConfig,
+      MetaDataParser metadataParser, DataSet<Vertex> vertices) {
+    List<TableToEdge> tablesToEdges = metadataParser.getTablesToEdges();
+    
+    DataSet<Edge> edges = null;
+    EdgeFactory edgeFactory = flinkConfig.getEdgeFactory();
+    ExecutionEnvironment env = flinkConfig.getExecutionEnvironment();
+
+    /*
+     * Foreign key relations to edges
+     */
+    DataSet<TableToEdge> dsTablesToEdges = env.fromCollection(tablesToEdges);
+
+    // Primary key table representation of foreign key relation
+    DataSet<LabelIdKeyTuple> pkEdges = dsTablesToEdges.filter(new EdgeDirectionFilter())
+        .flatMap(new VerticesToPkTable()).withBroadcastSet(vertices, "vertices").distinct("*");
```

Comment:
The name of broadcast variables should be a constant instead of an inline value.